src/cmds/qstat.c

```diff
@@ -302,49 +359,174 @@ states(char *string, char *q, char *r, char *h, char *w, char *t, char *e, int l
  * @param[in] n - attribute name
  * @param[in] r - resource name
  * @param[in] v - value for the attribute
+ * @param[in] one_line - one line per attribute
  *
  * @return Void
  *
  */
 void
-prt_attr(char *n, char *r, char *v)
+prt_attr(char *n, char *r, char *v, int one_line)
 {
-	char *c;
-	char *comma = ",";
-	int   first = 1;
-	int   l;
-	int   start;
-
-	start = strlen(n) + 7;	/* 4 spaces + ' = ' is 7 */
-	printf("    %s", n);
-	if (r) {
-		start += strlen(r) + 1;
-		printf(".%s", r);
-	}
-	printf(" = ");
-	c = strtok(v, comma);
-	while (c) {
-		if ((l = strlen(c)) + start < 78) {
-			printf("%s", c);
-			start += l;
-		} else {
-			if (! first) {
-				printf("\n\t");
-				start = 9;
+	int            first = 1;
+	int            l = 0;
+	int            start = 0;
+	int            i = 0;
+	char           *c = NULL;
+	char           *comma = ",";
+	static char    *prev_resc_name = NULL;
+	char           *buf = NULL;
+	char           *keyvalpair = NULL;
+	char           *key = NULL;
+	char           *val = NULL;
+	char           *temp = NULL;
+
+	if(v == NULL)
+		return;
+	switch (output_format) {
+		case FORMAT_JSON:
+			if (strcmp(n, ATTR_v) == 0) {
+				if (add_json_node(JSON_OBJECT, JSON_NULL, n, NULL) == NULL) {
+					fprintf(stderr, "qstat: out of memory\n");
+					exit(1);
+				}
+				keyvalpair = (char *) malloc(MAXPATHLEN);
+				l = MAXPATHLEN;
```

Comment:
We cannot restrict the user from using a long variable as long as the OS supports it. How about submitting a job using `-v` option and what if variables like `LS_COLORS` is present in the environment? I can find variables in my test VM of having the size of `1724` and no one is restricting me from putting more.
Regarding the variable names used, They were part of the legacy code and I reused them in my code.